dev/ToastNotifications/ToastNotificationManager.cpp

```diff
@@ -167,9 +169,76 @@ namespace winrt::Microsoft::Windows::ToastNotifications::implementation
 
         return static_cast<winrt::Microsoft::Windows::ToastNotifications::ToastNotificationSetting>(toastNotificationSetting);
     }
-    winrt::Microsoft::Windows::ToastNotifications::ToastNotificationHistory ToastNotificationManager::History()
+    winrt::Windows::Foundation::IAsyncAction ToastNotificationManager::RemoveWithIdentifierAsync(uint32_t toastIdentifier)
     {
-        throw hresult_not_implemented();
+        THROW_HR_IF(E_INVALIDARG, toastIdentifier == 0);
+
+        co_await winrt::resume_background();
+
+        std::wstring appId{ RetrieveToastAppId() };
+        ToastNotifications_RemoveToast(appId.c_str(), toastIdentifier);
+    }
+    winrt::Windows::Foundation::IAsyncAction ToastNotificationManager::RemoveWithTagAsync(hstring tag)
+    {
+        std::wstring _tag{ tag.c_str() };
+
+        co_await winrt::resume_background();
+
+        std::wstring appId{ RetrieveToastAppId() };
+        ToastNotifications_RemoveToastsWithTagAndGroup(appId.c_str(), _tag.c_str(), nullptr);
+    }
+    winrt::Windows::Foundation::IAsyncAction ToastNotificationManager::RemoveWithTagGroupAsync(hstring tag, hstring group)
+    {
+        std::wstring _tag{ tag.c_str() };
```

Comment:
Throw E_INVALIDARG if:
Tag or group is empty

Reply:
done